Managed identity cleanup

p3ck opened the pull request:
SUMMARY

This PR is finalize and cleanup work for Managed Identity feature.
This include the following changes:

    Remove "update_identities" definition and make sure all modules use "update_managed_identity" instead
    Remove "get new_identity" back off logic and make sure "new_identity" param is always provided ("new_identity" shouldn't be optional)
    Move all managed identity specs to "azure_rm_common_ext.py" to avoid duplicated specs , also move the relevant identity options description in doc_fragments
    For integration tests - consider using common tasks files for all managed identities preparation
    In integration tests - change all places using hard coded region to use '{{ location }}'

ISSUE TYPE

    Feature Pull Request

COMPONENT NAME

This Change is relevant for all modules that use the ManagedIdentity:

    plugins/modules/azure_rm_batchaccount.py
    plugins/modules/azure_rm_storageaccount.py
    plugins/modules/azure_rm_diskencryptionset.py
    plugins/modules/azure_rm_servicebus.py
    plugins/modules/azure_rm_recoveryservicesvault.py
    plugins/modules/azure_rm_cdnprofile.py
    plugins/modules/azure_rm_cosmosdbaccount.py
    plugins/modules/azure_rm_functionapp.py
    plugins/modules/azure_rm_eventhub.py
    plugins/modules/azure_rm_rediscache.py
    plugins/modules/azure_rm_appgateway.py
    plugins/modules/azure_rm_containerregistry.py
    plugins/modules/azure_rm_containerinstance.py
    plugins/modules/azure_rm_virtualmachinescaleset.py
    plugins/modules/azure_rm_sqlserver.py
    plugins/modules/azure_rm_postgresqlflexibleserver.py
    plugins/modules/azure_rm_webapp.py
    plugins/modules/azure_rm_iothub.py

nirarg committed:
Managed Identities: Remove "update_identities" definition

Change all places to use "update_managed_identity" function direct

nirarg committed:
update_managed_identity: change "new_identity" to mandatory param

Remove get new_identity back off logic and make sure new_identity param is always provided

nirarg committed:
Managed Identity: use common specs and doc fragments

Move all managed identity specs to azure_rm_common_ext.py to avoid duplicated specs
Also move the relevant identity descriptions to doc_fragments

nirarg committed:
Managed Identity: Move all identity preperation tasks to common file

Change all integration tests to use common tasks file for all managed identity operations
Also fix all hardcoded locations to the resource group location

p3ck commented:
@nirarg is on PTO now so I rebased his PR on latest dev and pushed it here.

xuzhang3 commented:
> @nirarg is on PTO now so I rebased his PR on latest dev and pushed it here.

working on it

Fred-sun commented on plugins/module_utils/azure_rm_common_ext.py:
```diff
@@ -281,17 +306,13 @@ def update_single_managed_identity(self, curr_identity, new_identity, patch_supp
             }
-        return self.update_managed_identity(curr_identity=curr_identity,
-                                            new_identity=new_identity_converted,
+        return self.update_managed_identity(new_identity=new_identity_converted,
```
This change will affect the  azure_rm_sqlmanagedinstance.py line 707

p3ck committed:
Update azure_rm_sqlmanagedinstance to use new update_managed_identity

This updates the azure_rm_sqlmanagedinstance module to use the new
arguments for the update_managed_identity method.

Fred-sun commented on plugins/module_utils/azure_rm_common_ext.py:
```diff
@@ -281,17 +306,13 @@ def update_single_managed_identity(self, curr_identity, new_identity, patch_supp
                                             allow_identities_append=False, patch_support=patch_support)
 
-    def update_managed_identity(self, curr_identity=None, new_identity=None,
+    def update_managed_identity(self, new_identity, curr_identity=None,
```
The arguments passed in the module call are all keyword arguments, and azure_rm_sqlserver.py 375 lines also need to be updated
```suggestion
    def update_managed_identity(self, new_identity={}, curr_identity=None,
```

Fred-sun reviewed (commented):
last changed request!

Fred-sun commented on tests/integration/targets/azure_rm_webapp/tasks/main.yml:
```diff
@@ -16,26 +16,22 @@
   ansible.builtin.set_fact:
-    user_identity_1: "/subscriptions/{{ azure_subscription_id }}/resourcegroups/{{ resource_group }}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/ansible-test-webapp-identity"
-    user_identity_2: "/subscriptions/{{ azure_subscription_id }}/resourcegroups/{{ resource_group }}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/ansible-test-webapp-identity-2"
+    location: "{{ __rg_info.resourcegroups.0.location }}"
```
Because the quota of eastus has been full before, eastus cannot be created. The web application and project should be in the same area (resource_group_third), thank you!
```suggestion
    location: westus2
```

Fred-sun commented on tests/integration/targets/azure_rm_webapp/tasks/main.yml:
```diff
@@ -16,26 +16,22 @@
+    managed_identity_test_unique: 'webapp'
+    managed_identity_unique: "{{ item }}"
+    managed_identity_action: 'create'
+    managed_identity_location: "{{ location }}"
```
```suggestion
    managed_identity_location: "{{ __rg_info.resourcegroups.0.location }}"
```

Fred-sun reviewed (commented):
The last changed request!

p3ck committed:
Update tests/integration/targets/azure_rm_webapp/tasks/main.yml

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>

p3ck committed:
Update tests/integration/targets/azure_rm_webapp/tasks/main.yml

Co-authored-by: Fred-sun <37327967+Fred-sun@users.noreply.github.com>
